File: src/org/proj/json/resource/JResourceEncoder.java
package org.proj.json.resource;

import org.proj.game.Resource;
import org.sjr.JSONObj;
import org.sjr.codec.JSONEncoder;

public class JResourceEncoder implements JSONEncoder<Resource> {
    final public static JResourceEncoder INSTANCE = new JResourceEncoder();
    private JResourceEncoder() {}

    @Override
    public JSONObj encode (Resource resource) {
        JSONObj resp = new JSONObj();
        resp.put("name", resource.getName());

        return resp;
    }
}

File: src/org/proj/json/resource/JResourceEncoder.kt
package org.proj.json.resource

import org.proj.game.Resource
import org.sjr.JSONObj
import org.sjr.codec.JSONEncoder

object JResourceEncoder: JSONEncoder<Resource> {
    override fun encode (entry: Resource): JSONObj {
        var response = JSONObj()
        response.put("name", entry.name)
        response.put("size", entry.size)
        response.put("probability", entry.probability)
        response.put("price", entry.price)

        return response
    }

    override fun getTargetClass(): Class<Resource> {
        return Resource::class.java
    }
}